File: skills/astra-toolkit/clients/csharp/examples/collections-insert-many-options.cs
using DataStax.AstraDB.DataApi;
using DataStax.AstraDB.DataApi.Collections;
using DataStax.AstraDB.DataApi.Core;

namespace Examples;

public class Program
{
  static async Task Main()
  {
    // Get an existing collection
    var client = new DataAPIClient();
    var database = client.GetDatabase(
      System.Environment.GetEnvironmentVariable("ASTRA_DB_API_ENDPOINT"),
      System.Environment.GetEnvironmentVariable("ASTRA_DB_APPLICATION_TOKEN")
    );
    var collection = database.GetCollection("**COLLECTION_NAME**");

    // Insert documents to the collection
    var document1 = new Document()
    {
      { "name", "Jane Doe" },
      { "age", 42 },
    };
    var document2 = new Document()
    {
      { "nickname", "Bobby" },
      { "color", "blue" },
      { "foods", new[] { "carrots", "chocolate" } },
    };
    var options = new CollectionInsertManyOptions()
    {
      ChunkSize = 2,
      Concurrency = 2,
      Ordered = false,
    };
    var result = await collection.InsertManyAsync(
      [document1, document2],
      options
    );

    foreach (var id in result.InsertedIds)
    {
      Console.WriteLine(id);
    }
  }
}

File: skills/astra-toolkit/clients/go/examples/collections-insert-many-options.go
package main

import (
	"context"
	"log"
	"os"
	"time"

	"github.com/datastax/astra-db-go/v2/astra"
	"github.com/datastax/astra-db-go/v2/astra/options"
)

func main() {
	ctx := context.Background()
	// Get an existing collection
	client := astra.NewClient()

	database := client.Database(
		os.Getenv("ASTRA_DB_API_ENDPOINT"),
		options.API().SetToken(os.Getenv("ASTRA_DB_APPLICATION_TOKEN")),
	)

	collection := database.Collection("**COLLECTION_NAME**")

	// Insert documents into the collection
	_, err := collection.InsertMany(
		ctx,
		[]map[string]any{
			{
				"name": "Jane Doe",
				"age":  42,
			},
			{
				"nickname": "Bobby",
				"color":    "blue",
				"foods":    []string{"carrots", "chocolate"},
			},
		},
		options.CollectionInsertMany().
			SetChunkSize(2).
			SetConcurrency(2).
			SetOrdered(false).
			UpdateAPIOptions(options.API().SetRequestTimeout(3*time.Second)),
	)
	if err != nil {
		log.Fatal(err)
	}
}
